### DISCUSSION TITLE
[2-1] this 포인터

### DISCUSSION THREAD

**Discussion by moonuckjoo** [bbang-jun/2023-OOP-Mentoring#8] [Q&A]
class Wonjoo {
private:
	int age;
	string name;
public:
	void setAge(int age) { this->age = age; }
	int getAge() { return this->age; }
	void setName(string name) { this->name = name; }
	string getName() { return this->name; }
이런 예제에서 뒤에 있는 this가 뭔지 모르겠습니다. 찾아본 결과 자기 자신의 객체를 가르킨다는데 그럼 메인함수에서 해당클래스를 통해서 만든 객체를 가르킨다는 뜻같기는 한데..
this->age = age;
이런 의미 자체를 모르겠습니다

**Comment by bbang-jun** [comment]
쉽게 말씀드리면 this->name의 의미는 현재 class가 있는 변수로 생각하시면 편하실 것 같습니다.

즉, private 안에 있는 name 변수에 setter 함수로 외부에서 가져온 데이터 값을 저장한다는 의미가 됩니다.

실시간 강의에서 지속적으로 설명 드린 부분이라서 이해가 안 되신다면 해당 부분을 녹화 강의로 수강 하시면 좋을 것 같습니다!
